spv/sync.go

```diff
@@ -1016,19 +1031,40 @@ func (s *Syncer) startupSync(ctx context.Context, rp *p2p.RemotePeer) error {
 					}
 					s.loadedFilters = true
 				}
+				if atomic.CompareAndSwapUint32(&s.atomicWalletSynced, 0, 1) &&
+					s.notificationCallbacks.SyncUpdated != nil {
+					s.notificationCallbacks.SyncUpdated(true)
+				}
 				return nil
 			}
+			// RescanPoint is != nil so we are not synced to the peer and
+			// check to see if it was previously synced
+			if atomic.CompareAndSwapUint32(&s.atomicWalletSynced, 1, 0) &&
+				s.notificationCallbacks.SyncUpdated != nil {
+				s.notificationCallbacks.SyncUpdated(false)
+			}
+
 			err = s.wallet.DiscoverActiveAddresses(ctx, rp, rescanPoint, s.discoverAccounts)
 			if err != nil {
 				return err
 			}
 			s.discoverAccounts = false
+			s.wallet.Lock()
```

Comment:
remove this, it's not the responsibility of the syncer to manage whether the wallet is locked or not.